File: 49-group-anagrams/group-anagrams.java
class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {
        HashMap< String, List<String>> map = new HashMap<>();
        
        for(String str : strs){
            char[] currArray = str.toCharArray();
            Arrays.sort(currArray);
            String key = new String(currArray);
            if(map.containsKey(key)){
                List<String> temp = map.get(key);
                temp.add(str);
                map.put(key,temp);
            }
            else{
                List<String> temp2 = new ArrayList<>();
                temp2.add(str);
                map.put(key, temp2);
            }
        }
        List<List<String>> result = new ArrayList<>(map.values());
        return result;
    }
}

File: 49-group-anagrams/group-anagrams.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        
        dictA = {}
        for s in strs:
            temp = ["0"]* 26
            for i in range(len(s)):
                temp[ord(s[i]) - 97] += "1"
            temp = "".join(temp)
            if temp in dictA:
                dictA[temp].append(s)
            else:
                dictA[temp] = [s]
        res = []
        for k,v in dictA.items():
            res.append(v)
        return res
        

        